codestodians/food-basket-front-end
# Test React Components with Jest

#### What does this PR do?
- It tests the proper rendering of React Components created so far.

#### Description of tasks completed?
- Upgrade of the babel preset used for transpiling.
- Upgrade of certain scripts in `package.json`.
- Installation of necessary packages for testing.
- Creation of a test helper to disable an unwanted React warning in the console while tests are running.
- Creation of render tests for the components.
- Refactoring of Import and export style of certain components.
- Autogeneration of snapshots as a result of testing.
- Inclusion of the coverage directory in `.gitignore`
- Deprecation of the User directory.
- Addition of an extra configuration option for the `HttpLink` in `App.jsx` to facilitate making requests.

#### How should this be manually tested?
- Check out to the `ft-setup-unit-testing-153397363` branch.
- Run `yarn install`
- Run tests with `yarn test` or `npm test` or `npm t`.

#### Any background context you want to provide?
- The pre-existing test command/instructions have been deprecated for the current one.
- This task was necessary to ensure that the coverage badge tells the correct story of where we are at as regards testing at the moment.
- Further test cases should subsequently be implemented so as to progressively build a proper testing history.

#### What are the relevant pivotal tracker stories?
- [#153397363](https://www.pivotaltracker.com/story/show/153397363)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A
